File: web/src/api/example/videoContentAnalysis.js
import service from '@/utils/request'

// 创建视频内容分析
export const createVideoContentAnalysis = (data) => {
  return service({
    url: '/tkhelper/video-analysis/create',
    method: 'post',
    data: data
  })
}

// 删除视频内容分析
export const deleteVideoContentAnalysis = (ID) => {
  return service({
    url: '/tkhelper/video-analysis/delete',
    method: 'delete',
    params: { ID }
  })
}

// 批量删除视频内容分析
export const deleteVideoContentAnalysisByIds = (IDs) => {
  return service({
    url: '/tkhelper/video-analysis/deleteByIds',
    method: 'delete',
    params: { 'IDs[]': IDs }
  })
}

// 更新视频内容分析
export const updateVideoContentAnalysis = (data) => {
  return service({
    url: '/tkhelper/video-analysis/update',
    method: 'put',
    data: data
  })
}

// 根据ID查询视频内容分析
export const findVideoContentAnalysis = (ID) => {
  return service({
    url: '/tkhelper/video-analysis/find',
    method: 'get',
    params: { ID }
  })
}

// 分页获取视频内容分析列表
export const getVideoContentAnalysisList = (params) => {
  return service({
    url: '/tkhelper/video-analysis/getVideoContentAnalysisList',
    method: 'get',
    params
  })
}

// 获取视频内容分析统计信息
export const getVideoContentAnalysisStatistics = () => {
  return service({
    url: '/tkhelper/video-analysis/getVideoContentAnalysisStatistics',
    method: 'get'
  })
}

// 从Excel导入视频内容分析数据
export const importVideoContentAnalysisFromExcel = (data) => {
  return service({
    url: '/tkhelper/video-analysis/importVideoContentAnalysisFromExcel',
    method: 'post',
    data: data
  })
}

// 批量更新处理状态
export const batchUpdateProcessStatus = (data) => {
  return service({
    url: '/tkhelper/video-analysis/batchUpdateProcessStatus',
    method: 'put',
    data: data
  })
}

// 不鉴权的视频内容分析接口
export const getVideoContentAnalysisPublic = () => {
  return service({
    url: '/tkhelper/video-analysis/getVideoContentAnalysisPublic',
    method: 'get'
  })
}

File: server/plugin/tkhelper/router/video_content_analysis.go
package router

import (
	"github.com/gin-gonic/gin"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/tkhelper/api"
)

type VideoContentAnalysisRouter struct{}

func (r *VideoContentAnalysisRouter) InitVideoContentAnalysisRouter(Router *gin.RouterGroup) {
	videoRouter := Router.Group("video-analysis")
	{
		videoRouter.POST("create", api.ApiGroupApp.VideoContentAnalysisApi.CreateVideoContentAnalysis)    // 创建视频分析记录
		videoRouter.DELETE("delete", api.ApiGroupApp.VideoContentAnalysisApi.DeleteVideoContentAnalysis) // 删除视频分析记录
		videoRouter.DELETE("deleteByIds", api.ApiGroupApp.VideoContentAnalysisApi.DeleteVideoContentAnalysisByIds) // 批量删除
		videoRouter.PUT("update", api.ApiGroupApp.VideoContentAnalysisApi.UpdateVideoContentAnalysis)    // 更新视频分析记录
		videoRouter.GET("find", api.ApiGroupApp.VideoContentAnalysisApi.FindVideoContentAnalysis)        // 根据ID获取视频分析记录
		videoRouter.GET("getVideoContentAnalysisList", api.ApiGroupApp.VideoContentAnalysisApi.GetVideoContentAnalysisList) // 获取视频分析记录列表

		// 新增的功能路由
		videoRouter.GET("getVideoContentAnalysisStatistics", api.ApiGroupApp.VideoContentAnalysisApi.GetVideoContentAnalysisStatistics) // 获取统计信息
		videoRouter.POST("importVideoContentAnalysisFromExcel", api.ApiGroupApp.VideoContentAnalysisApi.ImportVideoContentAnalysisFromExcel) // 从Excel导入
		videoRouter.PUT("batchUpdateProcessStatus", api.ApiGroupApp.VideoContentAnalysisApi.BatchUpdateProcessStatus) // 批量更新处理状态
		videoRouter.GET("getVideoContentAnalysisPublic", api.ApiGroupApp.VideoContentAnalysisApi.GetVideoContentAnalysisPublic) // 不鉴权接口
	}
}